test/nlp_program.jl

```diff
@@ -627,6 +632,125 @@ function test_compute_derivatives_Finite_Diff(;
     end
 end
 
+################################################
+#=
+# Test Objective Sensitivity wrt Parameters
+=#
+################################################
+
+function test_ObjectiveSensitivity()
+    # Model 1
+    model = Model(() -> DiffOpt.diff_optimizer(Ipopt.Optimizer))
+    set_silent(model)
+
+    # Parameters
+    @variable(model, p ∈ MOI.Parameter(1.5))
+    @variable(model, p_prox)
+
+    # Variables
+    @variable(model, x)
+
+    # Constraints
+    @constraint(model, con, p_prox == p) # dual fishing :)
+    @constraint(model, x * sin(p_prox) == 1)
+    @objective(model, Min, sum(x))
+
+    optimize!(model)
+    @assert is_solved_and_feasible(model)
+
+    # Set pertubations
+    Δp = 0.1
+    MOI.set(
+        model,
+        DiffOpt.ForwardConstraintSet(),
+        ParameterRef(p),
+        Parameter(Δp),
+    )
+
+    # Compute derivatives
+    DiffOpt.forward_differentiate!(model)
+
+    # Test Objective Sensitivity wrt parameters
+    df_dp = MOI.get(model, DiffOpt.ForwardObjectiveSensitivity())
+    @test isapprox(df_dp, dual(con) * Δp; atol = 1e-4)
```

Comment:
It would be nice to test the actual value as well, just to make sure we are not testing a bunch of zeros...